app/models/spree/flow_io_variant_decorator.rb

```diff
@@ -126,16 +128,64 @@ def price_in_zone(currency, product_zone)
     end
 
     def all_prices_in_zone(product_zone)
-      all_prices = prices.map { |price| { currency: price.currency, amount: (price.amount&.round || 0).to_s } }
+      all_prices = prices.map { |price| parse_price(price) }
 
-      flow_experience_key = product_zone&.flow_data&.[]('key')
+      flow_experience_key = product_zone&.flow_io_experience
       return all_prices if flow_experience_key.blank?
 
       flow_price = flow_local_price(flow_experience_key)
-      all_prices << { currency: flow_price.currency, amount: (flow_price.amount&.round || 0).to_s }
+      all_prices << parse_price(flow_price)
       all_prices
     end
 
+    def all_prices(zones)
+      countries = {}
+
+      zones.each do |zone|
+        all_prices = []
+
+        add_prices_for_available_currencies(zone, all_prices)
```

Comment:
is it ok for this method to modify the `all_prices` ? would it make more sense that the method returns the all_prices and that would be stored within the local all_prices? That way we could even be able to reuse this method if needed somewhere else?

Reply:
Good point! Changed!